### PR TITLE
Added more mutations and refactor test to test each mutation separately

### PR DESCRIPTION
**Pull request by makoscafee**
This PR fixes #

## Checklist
* [x] I have signed the CLA
* [x] I have updated/added any relevant documentation

### CODE REVIEW

File: pkg/validator/schema.go [hunk 1]
```diff
@@ -264,6 +271,7 @@ func applySchemaCheck(conf *config.Configuration, checkID string, test schemaTes
 		}
 		passes, issues, err = check.CheckContainer(test.Container)
 	} else {
+		prefix = "/spec"
```
Comment (rbren) [inline pkg/validator/schema.go:274]: I think this should be removed and put inside the check YAML

File: checks/memoryLimitsMissing.yaml [hunk 2]
```diff
@@ -25,3 +25,7 @@ schema:
               type: string
               not:
                 const: ''
+mutations:
+  - op: add
+    path: /resources/limits
```
Comment (rbren) [inline checks/memoryLimitsMissing.yaml:30]: shouldn't this be `/resources/limits/memory`?

File: checks/memoryRequestsMissing.yaml [hunk 3]
```diff
@@ -25,3 +25,7 @@ schema:
               type: string
               not:
                 const: ''
+mutations:
+  - op: add
+    path: /resources/requests
```
Comment (rbren) [inline checks/memoryRequestsMissing.yaml:30]: ditto

**Review by rbren — COMMENTED** [review]
For the two that appear to be broken, do we have a test case set up?

File: checks/memoryLimitsMissing.yaml [hunk 4]
```diff
@@ -25,3 +25,7 @@ schema:
...
+    value: {memory: "512Mi"}
```
Comment (rbren) [inline checks/memoryLimitsMissing.yaml:31]: Oh wait I get it now. Why are these different from CPU?

Comment (rbren) [reply checks/memoryLimitsMissing.yaml:31]: @makoscafee any reason these are different? I think this check would modify `limits.cpu` as-is

Comment (rbren) [reply checks/memoryLimitsMissing.yaml:31]: 🤔 maybe we need two mutations: one to add `limits`, and then one to add `limits.memory`?

Comment (makoscafee) [reply checks/memoryLimitsMissing.yaml]: Done

**Review by rbren — APPROVED** [review]

**Pull request merged**